### PR TITLE
Add practice exercise `connect`

### PR DESCRIPTION
**Pull request by jiegillet**
This one deals with graph traversal on a hexagonal lattice, a bit harder than average.

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -1080,7 +1080,7 @@
         "uuid": "594b6c1a-0a4c-4463-9535-62056c71404e",
         "practices": [
           "string",
-          "math"
+          "maths"
```
Comment (jiegillet) [inline config.json:1083]: I changed this for consistency, we had both math and maths. I picked math because it had more occurrences 🤷

Comment (lpil) [reply config.json:1083]: I've gone for the English spelling everywhere so maths is good by me 👍

File: exercises/practice/connect/.meta/example.gleam [hunk 2]
```diff
@@ -0,0 +1,152 @@
+import gleam/list
+import gleam/map.{Map}
+import gleam/string
+import gleam/result
+
+pub type Player {
+  X
+  O
+}
```
Comment (lpil) [inline exercises/practice/connect/.meta/example.gleam:9]: Not sure why but I find this type definition so satisfying

File: exercises/practice/connect/.meta/example.gleam [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  Size(rows: Int, columns: Int)
+}
+
+pub fn winner(board: String) -> Result(Player, Nil) {
+  let rows =
+    board
+    |> string.trim()
+    |> string.split("\n")
+
+  assert [first_row, ..] = rows
```
Comment (lpil) [inline exercises/practice/connect/.meta/example.gleam:25]: I would return an error here, the board could be empty.

**Review by lpil — APPROVED** [review]
Very nice! Looks like quite a tricky one.

I've left one small note

**Pull request merged**